docs/interaction/rpc.md

```diff
@@ -6,122 +6,675 @@ sidebar_label: RPC
 
-* `account/<account_id>` - returns view of account information, e.g. `{"amount": 1000000, "nonce": 102, "account_id": "test.near"}`
-* `access_key/<account_id>` - returns all access keys for given account.
-* `access_key/<account_id>/<public_key>` - returns details about access key for given account with this public key. If there is no such access key, returns nothing.
-* `contract/<account_id>` - returns full state of the contract \(might be expensive if contract has large state\).
-* `call/<account_id>/<method name>` - calls `<method name>` in contract `<account_id>` as view function with `data` as parameters.
 
-`http post http://127.0.0.1:3030/ jsonrpc=2.0 method=query params:="[\"account/test.near\",\"\"]" id="dontcare"`
+**`FullAccess` key**
+
+```bash
+http post https://rpc.nearprotocol.com jsonrpc=2.0 method=query params:='["access_key/near.test/ed25519:4Ynh6YmogjUYc5V9VqtQ1pRnZ4KuJSAJdkuj9hLj4a2p", ""]' id=dontcare
+```
 
-```javascript
+```json
 {
-    "id": "dontcare",
-    "jsonrpc": "2.0",
-    "result": {
-        "code": 0,
-        "codespace": "",
-        "height": 0,
-        "index": -1,
-        "info": "",
-        "key": "YWNjb3VudC90ZXN0Lm5lYXI=",
-        "log": "exists",
-        "proof": [],
-        "value": "eyJhY2NvdW50X2lkIjoidGVzdC5uZWFyIiwibm9uY2UiOjAsImFtb3VudCI6MTAwMDAwMDAwMDAwMCwic3Rha2UiOjUwMDAwMDAwLCJwdWJsaWNfa2V5cyI6W1sxNjIsMTIyLDE0MCwyMTksMTcyLDEwNSw4MCw3OCwxOTAsMTY1LDI1NSwxNDAsMTExLDQzLDIyLDE0OSwyMTEsMTUyLDIyNywyMjcsNjcsMjIyLDIzNCw3Nyw5NiwxNTYsNjYsMjMsMTcyLDk2LDc2LDEzN11dLCJjb2RlX2hhc2giOiJBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBPSJ9"
+  "id": "dontcare",
+  "jsonrpc": "2.0",
+  "result": {
+    "block_height": 288274,
+    "nonce": 1,
+    "permission": "FullAccess"
+  }
+}
+```
+
+**`FunctionCall` access key**
+
+```bash
+http post https://rpc.nearprotocol.com jsonrpc=2.0 method=query params:='["access_key/near.test/ed25519:9UfaMwBArFmccTYtzq43K3Kdoncp4kd8rx7cWk57oasL", ""]' id=dontcare
+```
+
+```json
+{
+  "id": "dontcare",
+  "jsonrpc": "2.0",
+  "result": {
+    "block_height": 288293,
+    "nonce": 0,
+    "permission": {
+      "FunctionCall": {
+        "allowance": "100000000",
+        "method_names": [],
+        "receiver_id": "studio-d4zxn4xdu"
+      }
     }
+  }
 }
 ```
 
-Where `value` is base58 encoded JSON of the account status: `'{"account_id":"test.near","nonce":0,"amount":1000000000000,"stake":50000000,"public_keys":[[...]],"code_hash":"AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA="}'`
+### `contract/<account_id>`
+
+Returns full state of the contract (might be expensive if contract has large state).
+
 
-the `[...]` will  be replaced with a public key as an array.
+### `call/<account_id>/<method name>`
 
-Note, this is Tendermint-like compatibility, that should be refactored.
+Calls `<method name>` in contract `<account_id>` as view function with `data` as parameters.
```

Comment:
The second parameter is a JSON string encoded with base58 (NOT base64):

```
$ http post https://rpc.nearprotocol.com/ jsonrpc=2.0 id=dontcare method=query 'params:=["call/studio-heptm2q29/whoSaidHi","AQ4"]'
{
    "id": "dontcare",
    "jsonrpc": "2.0",
    "result": {
        "block_height": 1045855,
        "logs": [],
        "result": [34, 102, 114, 111, 108, 46, 110, 101, 97, 114, 34]  // This is a JSON string: "frol.near" (with quotes)
    }
}
```

`AQ4` is `{}` (empty JSON object) in base58 encoding. The output result is an array of bytes (numbers) of JSON.

Reply:
✅